project/dexter-monitor/routes/project.js

```diff
@@ -71,6 +75,69 @@ exports.saveSnapshotSummary = function() {
         })
 };
 
+exports.createProject = function(req, res) {
+
+	Promise.resolve(req.body.project)
+	.then(createDexterServerDatabase)
+	.then(insertDexterMonitorEntry)
+	.then((project) => {
+		startDexterServer(project);
+		
+		//wait for the dexter server to start
+		setTimeout( () => {
+			log.info('Dexter Server instance created at:' + project.hostName + ":" + project.portNumber );
+			res.send({status:"ok"});
+		}, 5000);
+    })
+	 .catch((err) => {
+        log.error(`Failed to create Dexter Server instance : ${err.message}`);
+        res.send({status:"fail", errorMessage: err.message});
+    });
+}
+
+function createDexterServerDatabase(project) {
+	const sql = `CREATE DATABASE ${project.projectName}; USE ${project.projectName};`+
+			    fs.readFileSync(`${global.config.dexterServerPath}/config/ddl.sql`).toString()+
+			    `USE ${global.config.database.name};`;
+	return database.exec(sql).then(() => project);
+}
+
+function insertDexterMonitorEntry(project) {
+	const sql = `INSERT INTO ProjectInfo (projectName, pid, requester, administrator, portNumber, createdDateTime, language, hostIP, projectYear, emailingWhenServerDead, groupName, dbName) VALUES (?,?,?,?,?, now(), ?, ?, YEAR(now()), 'Y', '',?)`;
+	const args  = [project.projectName, project.projectName, project.adminName, project.adminName, project.portNumber, project.language, project.hostName, project.projectName];
```

Comment:
- The second argument seems to be a typo. `project.projectName` => `project.pid`
- The third and fourth arguments are same. Is it intended?